src/hyp3_isce2/insar_tops_multi_burst.py

```diff
@@ -0,0 +1,133 @@
+"""Create a single-burst Sentinel-1 geocoded unwrapped interferogram using ISCE2's TOPS processing workflow"""
+
+import argparse
+import logging
+import sys
+from typing import Iterable, Optional
+
+import isce  # noqa
+from burst2safe.burst2safe import burst2safe
+from hyp3lib.util import string_is_true
+from osgeo import gdal
+
+from hyp3_isce2.insar_tops import insar_tops_packaged
+from hyp3_isce2.insar_tops_burst import insar_tops_single_burst
+from hyp3_isce2.logger import configure_root_logger
+
+
+gdal.UseExceptions()
+
+log = logging.getLogger(__name__)
+
+
+def insar_tops_multi_burst(
+    reference: Iterable[str],
+    secondary: Iterable[str],
+    swaths: list = [1, 2, 3],
+    looks: str = '20x4',
+    apply_water_mask=False,
+    bucket: Optional[str] = None,
+    bucket_prefix: str = '',
+):
+    ref_ids = [g.split('_')[1] + '_' + g.split('_')[2] + '_' + g.split('_')[4] for g in reference]
+    sec_ids = [g.split('_')[1] + '_' + g.split('_')[2] + '_' + g.split('_')[4] for g in secondary]
+
+    if len(list(set(ref_ids) - set(sec_ids))) > 0:
```

Comment:
Was this the intention?
```suggestion
    if set(ref_ids) != set(sec_ids):
```